src/Middleware/HttpLogging/src/W3CLoggerProcessor.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class W3CLoggerProcessor : IDisposable
+    {
+        private const int _maxQueuedMessages = 1024;
+
+        private readonly string _path;
+        private readonly string _fileName;
+        private readonly int? _maxFileSize;
+        private readonly W3CLoggingFields _loggingFields;
+        private bool _hasWritten;
+
+        private readonly BlockingCollection<string> _messageQueue = new BlockingCollection<string>(_maxQueuedMessages);
```

Comment:
There's probably less ceremony here if you use a `BoundedChannel` with `BoundedChannelFullMode.DropWrite`, but I'll defer to others on what they recommend